examples/scala-video-base-with-data/README.md

```diff
@@ -0,0 +1,162 @@
+for your case should look similar to the posted below
+```json
+{"itemScores":[
+	{
+		"item":"i12",
+		"title":"title for movie i12",
+		"date":"1935",
+		"imdbUrl":"http://imdb.com/fake-url/i12",
+		"score":1.1700499715209998
+	},{
+		"item":"i44",
+		"title":"title for movie i44",
+		"date":"1974",
+		"imdbUrl":"http://imdb.com/fake-url/i44",
+		"score":1.1153550716504106
+	}
+]}
+``` 
+
+## SO, HOW TO?
+
+### The Main Idea
+
+Recall [the DASE Architecture](/start/engines/), a PredictionIO engine has
+4 main components: `Data Source`, `Data Preparator`, `Algorithm`, and `Serving`
+components. To achieve your goal, you will need provide the information about video to engine 
+(using sdk), and then let this information to pass from `Data Source` through all the engine
+to the `Serving` component where the engine will send required information back to your application.
+
+### Implementation
+
+#### Modify The Item
+In file [DataSource.scala](https://github.com/PredictionIO/PredictionIO/blob/develop/examples/scala-parallel-similarproduct-multi/src/main/scala/DataSource.scala)
+you will find class `Item` defined in the next way
+```scala
+case class Item(categories: Option[List[String]])
+```
+
+At the first, we need simply add required fields to this class
+```scala
+case class Item(
+	title: String,
+	date: String,
+	imdbUrl: String,
+	categories: Option[List[String]])
+```
+
+#### Create The Item Properly
+Now, your IDE (or compiler) will say you about all the places where you need make changes to create item
+properly. For example, [DataSource.scala#L50](https://github.com/PredictionIO/PredictionIO/blob/develop/examples/add-scala-video-base-with-data/examples/scala-video-base-with-data/src/main/scala/DataSource.scala#L50)
+```scala
+Item(categories = properties.getOpt[List[String]]("categories"))
+```
+You need now to add needed properties to item
+```scala
+Item(
+	title = properties.get[String]("title"),
+	date = properties.get[String]("date"),
+	imdbUrl = properties.get[String]("imdbUrl"),
+	categories = properties.getOpt[List[String]]("categories"))
+```
+
+#### Modify The ItemScore
+Now, when you've fixed item creation, take a look on class `ItemScore` from the file [Engine.scala](https://github.com/PredictionIO/PredictionIO/blob/develop/examples/scala-parallel-similarproduct-multi/src/main/scala/Engine.scala)
+```scala
+case class ItemScore(
+	item: String,
+	score: Double
+) extends Serializable
+```
+Engine will return class `PredictedResult` which contains property `itemScores: Array[ItemScore]`.
+So, since your result is of class`ItemScore`, you need modify it too. 
+In our example after modification you will have something similar to below
+```scala 
+case class ItemScore(
+	item: String,
+	title: String,
+	date: String,
+	imdbUrl: String,
+	score: Double
+) extends Serializable
+```
+
+And again now you need go through all the places where `ItemScore` is created and fix compiler errors.
```

Comment:
Hi, would you elaborate more here what else needs to be modified (like how you describe changes for the case classes)? Thanks!